sklearn/mixture/_base.py

```diff
@@ -234,7 +234,7 @@ def fit_predict(self, X, y=None):
 
             self._print_verbose_msg_init_end(lower_bound)
 
-            if lower_bound > max_lower_bound:
+            if lower_bound > max_lower_bound or max_lower_bound == -np.inf:
```

Comment:
This is to fix another error. Please check [this](https://github.com/scikit-learn/scikit-learn/issues/18216)

Reply:
Thanks for providing the issue.

As this PR introduces a new feature, it would better fixing #18216 first and independently in another pull request.

Reply:
Sure! I will make another PR then!

Reply:
> Sure! I will make another PR then!

Feel free to add me as a reviewer. :wink: